### PR TITLE
DLL build

### PR DESCRIPTION
**Pull request by assafnativ**
Added VisualStudio project for building Detours as a DLL.

### CODE REVIEW

File: build/userLand.props [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    </ClCompile>
+  </ItemDefinitionGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)'=='Debug'" Label="DebugOnly">
+    <ClCompile>
+      <PreprocessorDefinitions>_ITERATOR_DEBUG_LEVEL=2;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+    </ClCompile>
+  </ItemDefinitionGroup>
+  <ItemDefinitionGroup>
+    <ClCompile>
+      <PreprocessorDefinitions>_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING;NOMINMAX;USERLAND=1;GTEST_HAS_TR1_TUPLE=0;GTEST_USE_OWN_TR1_TUPLE=0;GTEST_LANG_CXX11=1;RPC_USE_NATIVE_WCHAR;_ATL_NO_WIN_SUPPORT;_ATL_NO_AUTOMATIC_NAMESPACE;_ATL_CSTRING_EXPLICIT_CONSTRUCTORS;_NO_PPLXIMP;_NO_ASYNCRTIMP;_WINSOCK_DEPRECATED_NO_WARNINGS;_SCL_SECURE_NO_WARNINGS;WIN32;_WINDOWS;_UNICODE;UNICODE;STRICT;STRICT_TYPED_ITEMIDS;%(PreprocessorDefinitions)</PreprocessorDefinitions>
```
[the remark this answers is not in the document]
Comment (assafnativ) [reply build/userLand.props]: Just some stuff I've copied from another project. I'll remove them.

File: build/userLand.props [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      <BasicRuntimeChecks>Default</BasicRuntimeChecks>
+      <BufferSecurityCheck>false</BufferSecurityCheck>
+      <ExceptionHandling>false</ExceptionHandling>
+      <ControlFlowGuard>false</ControlFlowGuard>
+      <IntrinsicFunctions>false</IntrinsicFunctions>
+      <WholeProgramOptimization>false</WholeProgramOptimization>
+    </ClCompile>
+    <Link>
+      <AdditionalLibraryDirectories Condition="'$(Platform)'=='Win32'">$(SolutionDir)WscIsvApi\Win32\;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
+      <AdditionalLibraryDirectories Condition="'$(Platform)'=='x64'">$(SolutionDir)WscIsvApi\amd64\;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
```
[the remark this answers is not in the document]
Comment (assafnativ) [reply build/userLand.props]: Again, copy/past mistake.

File: build/sdk.props [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <MyLatestTargetPlatformVersion Condition="'$(MyLatestTargetPlatformVersion)' == ''">$([Microsoft.Build.Utilities.ToolLocationHelper]::GetLatestSDKTargetPlatformVersion('Windows', '10.0'))</MyLatestTargetPlatformVersion>
+    <WindowsTargetPlatformVersion Condition="'$(WindowsTargetPlatformVersion)' == ''">$(MyLatestTargetPlatformVersion)</WindowsTargetPlatformVersion>
```
[the remark this answers is not in the document]
Comment (assafnativ) [reply build/sdk.props]: PlatformVersion is not defined anywhere. There were issues in the past in which this param was not auto set correctly.

[the remark this answers is not in the document]
Comment (sylveon) [reply build/sdk.props]: You missed my point.
```xml
<MyLatestTargetPlatformVersion Condition="'$(MyLatestTargetPlatformVersion)' == ''">$([Microsoft.Build.Utilities.ToolLocationHelper]::GetLatestSDKTargetPlatformVersion('Windows', '10.0'))</MyLatestTargetPlatformVersion>
````
is unneeded when you can just do
```xml
<WindowsTargetPlatformVersion Condition="'$(WindowsTargetPlatformVersion)' == ''">10.0</WindowsTargetPlatformVersion>
```

[the remark this answers is not in the document]
Comment (assafnativ) [reply build/sdk.props]: There was a reason for that, but I can't recall what it was, so I'll change it to what you are saying.

**Comment by dtarditi** [comment]
We do not recommend distributing Detours as a DLL.  See this entry in the FAQ for the reasons why: https://github.com/microsoft/detours/wiki/FAQ#why-is-detours-packaged-as-a-static-library-detourslib-and-not-as-a-dynamic-link-library-say-detoursdll.
It is too easy to step on Detours itself.

**Comment by dtarditi** [comment]
Closing per feedback on distributing Detours.

**Pull request closed without merging**